fdbserver/storageserver.actor.cpp

```diff
@@ -4704,14 +4697,31 @@ ACTOR Future<GetMappedKeyValuesReply> mapKeyValues(StorageServer* data,
 			                      "storageserver.mapKeyValues.AfterBatch");
 		subqueries.clear();
 		for (int i = 0; i + offset < sz && i < SERVER_KNOBS->MAX_PARALLEL_QUICK_GET_VALUE; i++) {
-			result.data.push_back(result.arena, kvms[i]);
+			int size = sizeof(KeyValueRef) + input.data[i + offset].expectedSize() + getMappedKeyValueSize(kvms[i]);
+			// allow at least 1 entry to be returned
+			if (SERVER_KNOBS->STRICTLY_ENFORCE_BYTE_LIMIT && *remainingLimitBytes < size &&
+			    !(firstBatch && i + offset == 0)) {
+				// clear the remainingLimitBytes to quit from the outer loop
+				*remainingLimitBytes = 0;
+				break;
+			}
+			*remainingLimitBytes -= size;
+			++cnt;
+			result->data.push_back(result->arena, kvms[i]);
+
+			// fill index KV for open boundary
+			if (firstBatch && i + offset == 0) {
+				result->data[0].key = input.data[0].key;
+				result->data[0].value = input.data[0].value;
+			}
```

Comment:
we provide the index key for both the begin and the end of the range, so that client can use them as continuations -- I think it is because we might scan forward/backward, so we add both. Here it is the beginning.